File: Ent_UsAdmin/Creditos/CreditosAsociadosActualizarCliente.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class CreditosAsociadosActualizarCliente
    {
        public DataTable dataCreditos;
        public string nombre;
        public string aPaterno;
        public string aMaterno;
        public string Curp;
        public string telefono;
        public string Celular;
        public string valorAnterior;
        public string idCliente;
        public string EstadoValidacionCurp;
        public bool ValidadoCurp;
        public DateTime FechaNacimiento;
        public bool Autorizado;
        private string idActualizacion;
        private bool err;
        private Cargando ncargando;
        public bool creada;
        private string existeActualizacion;

        public CreditosAsociadosActualizarCliente()
        {
            InitializeComponent();
        }
        private void CreditosAsociadosActualizarCliente_Load(object sender, EventArgs e)
        {

            dtdatos.DataSource = dataCreditos;
            btn_agregar.Enabled = false;

            ncargando = new Cargando();
            ncargando.MonoFlat_Label1.Text = "Consultando información";
            ncargando.Show();

            BackgroundConsultarActualizacion.RunWorkerAsync();


        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void btn_agregar_Click(object sender, EventArgs e)
        {
            btn_agregar.Enabled = false;

            foreach (DataRow row in Module1.dataPermisos.Rows)
            {
                if (Conversions.ToBoolean(row["SatiModCreditosModificar"]))
                {
                    My.MyProject.Forms.Autorizacion.tipoAutorizacion = "SatiModCreditosModificar";
                    My.MyProject.Forms.Autorizacion.ShowDialog();
                    if (Autorizado)
                    {
                        ncargando = new Cargando();

                        ncargando.MonoFlat_Label1.Text = "Actualizando cliente";
                        ncargando.Show();
                        BackgroundActualizar.RunWorkerAsync();
                    }
                    else
                    {
                        MessageBox.Show("No fue autorizado");
                    }
                    break;
                }
                else if (MessageBox.Show("¿No cuenta con los permisos, desea notificar a un usuario?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
                {
                    ncargando = new Cargando();
                    ncargando.MonoFlat_Label1.Text = "Creando actualización";
                    ncargando.Show();
                    BackgroundActualizarCnotificacion.RunWorkerAsync();


                }
            }
        }

        private void BackgroundActualizar_DoWork(object sender, DoWorkEventArgs e)
        {
            try
            {
                Module1.iniciarconexionempresa();

                SqlCommand comandoActualizar;
                string consultaActualizar;

                consultaActualizar = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(@"INSERT INTO [dbo].[ActualizacionesClientes]
          
     VALUES
           (" + idCliente + @"
           ,'" + nombre + @"'
           ,'" + aPaterno + @"'
           ,'" + aMaterno + @"'
           ,'" + Curp + @"'
           ,'" + EstadoValidacionCurp + @"'
           ,'" + ValidadoCurp + @"'
           ,'" + telefono + @"'
           ,'" + Celular + @"'
           ,'" + FechaNacimiento.ToString("yyyy-MM-dd") + @"'
           ,'", dtdatos.Rows[dtdatos.CurrentRow.Index].Cells[0].Value), @"'
           ,'"), valorAnterior), @"'
           ,GETDATE()
           ,GETDATE()
           ,'"), Module1.nmusr), @"'
           ,''
           ,''
           ,''
           ,'E','0') Select SCOPE_IDENTITY()"));
                comandoActualizar = new SqlCommand();
                comandoActualizar.Connection = Module1.conexionempresa;
                comandoActualizar.CommandText = consultaActualizar;
                idActualizacion = Conversions.ToString(comandoActualizar.ExecuteScalar());

                SqlCommand comandoProdActualizar;
                string consultaProdActualizar;
                consultaProdActualizar = "ActualizarCliente";
                comandoProdActualizar = new SqlCommand();
                comandoProdActualizar.Connection = Module1.conexionempresa;
                comandoProdActualizar.CommandText = consultaProdActualizar;

                comandoProdActualizar.CommandType = CommandType.StoredProcedure;
                comandoProdActualizar.Parameters.AddWithValue("idActualizacion", idActualizacion);
                comandoProdActualizar.Parameters.AddWithValue("usr", Module1.nmusr);
                comandoProdActualizar.ExecuteNonQuery();
                err = false;
            }


            catch (Exception ex)
            {
                err = true;
                MessageBox.Show(ex.Message);
            }



        }

        private void BackgroundActualizar_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();

            if (err)
            {
                MessageBox.Show("Hubo un error inténtalo de nuevo");
            }
            else
            {
                MessageBox.Show("Actualizado con éxito");
                Close();

            }
            btn_agregar.Enabled = true;

        }

        private void BackgroundActualizarCnotificacion_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();

            SqlCommand comandoActualizar;
            string consultaActualizar;

            consultaActualizar = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(@"INSERT INTO [dbo].[ActualizacionesClientes]
          
     VALUES  (" + idCliente + ",'" + nombre + "','" + aPaterno + "','" + aMaterno + "','" + Curp + "','" + EstadoValidacionCurp + "','" + ValidadoCurp + "','" + telefono + "','" + Celular + "','" + FechaNacimiento.ToString("yyyy-MM-dd") + "','", dtdatos.Rows[dtdatos.CurrentRow.Index].Cells[0].Value), "','"), valorAnterior), "',GETDATE(),GETDATE(),'"), Module1.nmusr), "','','','','E','0'); Select SCOPE_IDENTITY()"));
            comandoActualizar = new SqlCommand();
            comandoActualizar.Connection = Module1.conexionempresa;
            comandoActualizar.CommandText = consultaActualizar;
            idActualizacion = Conversions.ToString(comandoActualizar.ExecuteScalar());
        }

        private void BackgroundActualizarCnotificacion_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();
            My.MyProject.Forms.CrearNotificacionActualizarCliente.idActualizacion = idActualizacion;
            // CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
            My.MyProject.Forms.CrearNotificacionActualizarCliente.ShowDialog();

            if (creada)
            {
                Task.Factory.StartNew(() =>
                    {
                        SqlCommand comandoActualizaPromesa;
                        string consultaActualizaPromesa;
                        consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" + idActualizacion + "'";
                        comandoActualizaPromesa = new SqlCommand();
                        comandoActualizaPromesa.Connection = Module1.conexionempresa;
                        comandoActualizaPromesa.CommandText = consultaActualizaPromesa;
                        comandoActualizaPromesa.ExecuteNonQuery();


                    });
                Close();
            }
            else if (MessageBox.Show("¿La notificación no fue creada, desea volver a intentarlo?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
            {
                My.MyProject.Forms.CrearNotificacionActualizarCliente.idActualizacion = idActualizacion;
                // CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
                My.MyProject.Forms.CrearNotificacionActualizarCliente.ShowDialog();
                if (creada)
                {
                    Task.Factory.StartNew(() =>
                    {
                        SqlCommand comandoActualizaPromesa;
                        string consultaActualizaPromesa;
                        consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" + idActualizacion + "'";
                        comandoActualizaPromesa = new SqlCommand();
                        comandoActualizaPromesa.Connection = Module1.conexionempresa;
                        comandoActualizaPromesa.CommandText = consultaActualizaPromesa;
                        comandoActualizaPromesa.ExecuteNonQuery();


                    });
                    Close();
                }
                else
                {
                    MessageBox.Show("La notificación no fue creada, puedes intentarlo más tarde");
                    Close();

                }
            }
            else
            {
                Close();

            }
            btn_agregar.Enabled = true;


        }

        private void BackgroundConsultarActualizacion_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();
            string consultaActualizacion;
            consultaActualizacion = " if exists(Select * from actualizacionesClientes where idcliente = '" + idCliente + @"' and estado = 'E' and notificado = 0)
							begin
							 select 'Pendiente'
							end
							else if exists(select * from actualizacionesClientes where idcliente = '" + idCliente + @"' and estado = 'E' and Notificado = 1)
                            begin
                             Select 'Notificado'
                            end
                            else
							select 'Noexiste'
";


            SqlCommand comandoActualizacion;
            comandoActualizacion = new SqlCommand();
            comandoActualizacion.Connection = Module1.conexionempresa;
            comandoActualizacion.CommandText = consultaActualizacion;
            existeActualizacion = Conversions.ToString(comandoActualizacion.ExecuteScalar());
        }

        private void BackgroundConsultarActualizacion_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();

            if (existeActualizacion == "Existe")
            {

                MessageBox.Show("El crédito ya cuenta con una actualización activa");
                Close();
            }
            else if (existeActualizacion == "Pendiente")
            {
                MessageBox.Show("El cliente cuenta con una actualización pendiente de notificar, se intentará notificar de nuevo");

                ncargando = new Cargando();
                ncargando.Show();
                ncargando.MonoFlat_Label1.Text = "Consultando Actualización";
                ncargando.TopMost = true;
                BackgroundConsultarActualizacionPendiente.RunWorkerAsync();
            }

            else if (existeActualizacion == "Notificado")
            {
                MessageBox.Show("El cliente cuenta con una notificación pendiente de autorizar");
                Close();
            }
            else
            {
                btn_agregar.Enabled = true;


            }
        }

        private void BackgroundConsultarActualizacionPendiente_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexionempresa();
            string consultaActualizacion;
            consultaActualizacion = "select id from actualizacionesClientes where idcliente = '" + idCliente + "' and estado = 'E' and notificado = 0";
            SqlCommand comandoActualizacion;
            comandoActualizacion = new SqlCommand();
            comandoActualizacion.Connection = Module1.conexionempresa;
            comandoActualizacion.CommandText = consultaActualizacion;
            idActualizacion = Conversions.ToString(comandoActualizacion.ExecuteScalar());
        }

        private void BackgroundConsultarActualizacionPendiente_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();
            My.MyProject.Forms.CrearNotificacionActualizarCliente.idActualizacion = idActualizacion;
            // CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
            My.MyProject.Forms.CrearNotificacionActualizarCliente.ShowDialog();

            if (creada)
            {
                Task.Factory.StartNew(() =>
                    {
                        SqlCommand comandoActualizaPromesa;
                        string consultaActualizaPromesa;
                        consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" + idActualizacion + "'";
                        comandoActualizaPromesa = new SqlCommand();
                        comandoActualizaPromesa.Connection = Module1.conexionempresa;
                        comandoActualizaPromesa.CommandText = consultaActualizaPromesa;
                        comandoActualizaPromesa.ExecuteNonQuery();


                    });
                Close();
            }
            else if (MessageBox.Show("¿La notificación no fue creada, desea volver a intentarlo?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
            {
                My.MyProject.Forms.CrearNotificacionActualizarCliente.idActualizacion = idActualizacion;
                // CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
                My.MyProject.Forms.CrearNotificacionActualizarCliente.ShowDialog();
                if (creada)
                {
                    Task.Factory.StartNew(() =>
                    {
                        SqlCommand comandoActualizaPromesa;
                        string consultaActualizaPromesa;
                        consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" + idActualizacion + "'";
                        comandoActualizaPromesa = new SqlCommand();
                        comandoActualizaPromesa.Connection = Module1.conexionempresa;
                        comandoActualizaPromesa.CommandText = consultaActualizaPromesa;
                        comandoActualizaPromesa.ExecuteNonQuery();


                    });
                    Close();
                }
                else
                {
                    MessageBox.Show("La notificación no fue creada, puedes intentarlo más tarde");
                    Close();

                }
            }
            else
            {
                Close();

            }
        }
    }
}

File: Ent_UsAdmin/CreditosAsociadosActualizarCliente.vb
Imports System.ComponentModel
Imports System.Data.SqlClient
Imports System.Threading.Tasks

Public Class CreditosAsociadosActualizarCliente
    Public dataCreditos As DataTable
    Public nombre As String
    Public aPaterno As String
    Public aMaterno As String
    Public Curp As String
    Public telefono As String
    Public Celular As String
    Public valorAnterior As String
    Public idCliente As String
    Public EstadoValidacionCurp As String
    Public ValidadoCurp As Boolean
    Public FechaNacimiento As Date
    Public Autorizado As Boolean
    Dim idActualizacion As String
    Dim err As Boolean
    Dim ncargando As Cargando
    Public creada As Boolean
    Dim existeActualizacion As String
    Private Sub CreditosAsociadosActualizarCliente_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        dtdatos.DataSource = dataCreditos
        btn_agregar.Enabled = False

        ncargando = New Cargando
        ncargando.MonoFlat_Label1.Text = "Consultando información"
        ncargando.Show()

        BackgroundConsultarActualizacion.RunWorkerAsync()


    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub btn_agregar_Click(sender As Object, e As EventArgs) Handles btn_agregar.Click
        btn_agregar.Enabled = False

        For Each row As DataRow In dataPermisos.Rows
            If row("SatiModCreditosModificar") Then
                Autorizacion.tipoAutorizacion = "SatiModCreditosModificar"
                Autorizacion.ShowDialog()
                If Autorizado Then
                    ncargando = New Cargando

                    ncargando.MonoFlat_Label1.Text = "Actualizando cliente"
                    ncargando.Show()
                    BackgroundActualizar.RunWorkerAsync()
                Else
                    MessageBox.Show("No fue autorizado")
                End If
                Exit For
            Else
                If MessageBox.Show("¿No cuenta con los permisos, desea notificar a un usuario?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                    ncargando = New Cargando
                    ncargando.MonoFlat_Label1.Text = "Creando actualización"
                    ncargando.Show()
                    BackgroundActualizarCnotificacion.RunWorkerAsync()


                End If
            End If
        Next
    End Sub

    Private Sub BackgroundActualizar_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundActualizar.DoWork
        Try
            iniciarconexionempresa()

            Dim comandoActualizar As SqlCommand
            Dim consultaActualizar As String

            consultaActualizar = "INSERT INTO [dbo].[ActualizacionesClientes]
          
     VALUES
           (" & idCliente & "
           ,'" & nombre & "'
           ,'" & aPaterno & "'
           ,'" & aMaterno & "'
           ,'" & Curp & "'
           ,'" & EstadoValidacionCurp & "'
           ,'" & ValidadoCurp & "'
           ,'" & telefono & "'
           ,'" & Celular & "'
           ,'" & FechaNacimiento.ToString("yyyy-MM-dd") & "'
           ,'" & dtdatos.Rows(dtdatos.CurrentRow.Index).Cells(0).Value & "'
           ,'" & valorAnterior & "'
           ,GETDATE()
           ,GETDATE()
           ,'" & nmusr & "'
           ,''
           ,''
           ,''
           ,'E','0') Select SCOPE_IDENTITY()"
            comandoActualizar = New SqlCommand
            comandoActualizar.Connection = conexionempresa
            comandoActualizar.CommandText = consultaActualizar
            idActualizacion = comandoActualizar.ExecuteScalar

            Dim comandoProdActualizar As SqlCommand
            Dim consultaProdActualizar As String
            consultaProdActualizar = "ActualizarCliente"
            comandoProdActualizar = New SqlCommand
            comandoProdActualizar.Connection = conexionempresa
            comandoProdActualizar.CommandText = consultaProdActualizar

            comandoProdActualizar.CommandType = CommandType.StoredProcedure
            comandoProdActualizar.Parameters.AddWithValue("idActualizacion", idActualizacion)
            comandoProdActualizar.Parameters.AddWithValue("usr", nmusr)
            comandoProdActualizar.ExecuteNonQuery()
            err = False


        Catch ex As Exception
            err = True
            MessageBox.Show(ex.Message)
        End Try



    End Sub

    Private Sub BackgroundActualizar_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundActualizar.RunWorkerCompleted
        ncargando.Close()

        If err Then
            MessageBox.Show("Hubo un error inténtalo de nuevo")
        Else
            MessageBox.Show("Actualizado con éxito")
            Me.Close()

        End If
        btn_agregar.Enabled = True

    End Sub

    Private Sub BackgroundActualizarCnotificacion_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundActualizarCnotificacion.DoWork
        iniciarconexionempresa()

        Dim comandoActualizar As SqlCommand
        Dim consultaActualizar As String

        consultaActualizar = "INSERT INTO [dbo].[ActualizacionesClientes]
          
     VALUES  (" & idCliente & ",'" & nombre & "','" & aPaterno & "','" & aMaterno & "','" & Curp & "','" & EstadoValidacionCurp & "','" & ValidadoCurp & "','" & telefono & "','" & Celular & "','" & FechaNacimiento.ToString("yyyy-MM-dd") & "','" & dtdatos.Rows(dtdatos.CurrentRow.Index).Cells(0).Value & "','" & valorAnterior & "',GETDATE(),GETDATE(),'" & nmusr & "','','','','E','0'); Select SCOPE_IDENTITY()"
        comandoActualizar = New SqlCommand
        comandoActualizar.Connection = conexionempresa
        comandoActualizar.CommandText = consultaActualizar
        idActualizacion = comandoActualizar.ExecuteScalar
    End Sub

    Private Sub BackgroundActualizarCnotificacion_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundActualizarCnotificacion.RunWorkerCompleted
        ncargando.Close()
        CrearNotificacionActualizarCliente.idActualizacion = idActualizacion
        '  CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
        CrearNotificacionActualizarCliente.ShowDialog()

        If creada Then
            Task.Factory.StartNew(Sub()
                                      Dim comandoActualizaPromesa As SqlCommand
                                      Dim consultaActualizaPromesa As String
                                      consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" & idActualizacion & "'"
                                      comandoActualizaPromesa = New SqlCommand
                                      comandoActualizaPromesa.Connection = conexionempresa
                                      comandoActualizaPromesa.CommandText = consultaActualizaPromesa
                                      comandoActualizaPromesa.ExecuteNonQuery()


                                  End Sub)
            Me.Close()
        Else
            If MessageBox.Show("¿La notificación no fue creada, desea volver a intentarlo?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                CrearNotificacionActualizarCliente.idActualizacion = idActualizacion
                'CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
                CrearNotificacionActualizarCliente.ShowDialog()
                If creada Then
                    Task.Factory.StartNew(Sub()
                                              Dim comandoActualizaPromesa As SqlCommand
                                              Dim consultaActualizaPromesa As String
                                              consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" & idActualizacion & "'"
                                              comandoActualizaPromesa = New SqlCommand
                                              comandoActualizaPromesa.Connection = conexionempresa
                                              comandoActualizaPromesa.CommandText = consultaActualizaPromesa
                                              comandoActualizaPromesa.ExecuteNonQuery()


                                          End Sub)
                    Me.Close()
                Else
                    MessageBox.Show("La notificación no fue creada, puedes intentarlo más tarde")
                    Me.Close()

                End If
            Else
                Me.Close()
            End If

        End If
        btn_agregar.Enabled = True


    End Sub

    Private Sub BackgroundConsultarActualizacion_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundConsultarActualizacion.DoWork
        iniciarconexionempresa()
        Dim consultaActualizacion As String
        consultaActualizacion = " if exists(Select * from actualizacionesClientes where idcliente = '" & idCliente & "' and estado = 'E' and notificado = 0)
							begin
							 select 'Pendiente'
							end
							else if exists(select * from actualizacionesClientes where idcliente = '" & idCliente & "' and estado = 'E' and Notificado = 1)
                            begin
                             Select 'Notificado'
                            end
                            else
							select 'Noexiste'
"


        Dim comandoActualizacion As SqlCommand
        comandoActualizacion = New SqlCommand
        comandoActualizacion.Connection = conexionempresa
        comandoActualizacion.CommandText = consultaActualizacion
        existeActualizacion = comandoActualizacion.ExecuteScalar
    End Sub

    Private Sub BackgroundConsultarActualizacion_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundConsultarActualizacion.RunWorkerCompleted
        ncargando.Close()

        If existeActualizacion = "Existe" Then

            MessageBox.Show("El crédito ya cuenta con una actualización activa")
            Me.Close()
        ElseIf existeActualizacion = "Pendiente" Then
            MessageBox.Show("El cliente cuenta con una actualización pendiente de notificar, se intentará notificar de nuevo")

            ncargando = New Cargando
            ncargando.Show()
            ncargando.MonoFlat_Label1.Text = "Consultando Actualización"
            ncargando.TopMost = True
            BackgroundConsultarActualizacionPendiente.RunWorkerAsync()

        ElseIf existeActualizacion = "Notificado" Then
            MessageBox.Show("El cliente cuenta con una notificación pendiente de autorizar")
            Me.Close()
        Else
            btn_agregar.Enabled = True


        End If
    End Sub

    Private Sub BackgroundConsultarActualizacionPendiente_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundConsultarActualizacionPendiente.DoWork
        iniciarconexionempresa()
        Dim consultaActualizacion As String
        consultaActualizacion = "select id from actualizacionesClientes where idcliente = '" & idCliente & "' and estado = 'E' and notificado = 0"
        Dim comandoActualizacion As SqlCommand
        comandoActualizacion = New SqlCommand
        comandoActualizacion.Connection = conexionempresa
        comandoActualizacion.CommandText = consultaActualizacion
        idActualizacion = comandoActualizacion.ExecuteScalar
    End Sub

    Private Sub BackgroundConsultarActualizacionPendiente_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundConsultarActualizacionPendiente.RunWorkerCompleted
        ncargando.Close()
        CrearNotificacionActualizarCliente.idActualizacion = idActualizacion
        '  CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
        CrearNotificacionActualizarCliente.ShowDialog()

        If creada Then
            Task.Factory.StartNew(Sub()
                                      Dim comandoActualizaPromesa As SqlCommand
                                      Dim consultaActualizaPromesa As String
                                      consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" & idActualizacion & "'"
                                      comandoActualizaPromesa = New SqlCommand
                                      comandoActualizaPromesa.Connection = conexionempresa
                                      comandoActualizaPromesa.CommandText = consultaActualizaPromesa
                                      comandoActualizaPromesa.ExecuteNonQuery()


                                  End Sub)
            Me.Close()
        Else
            If MessageBox.Show("¿La notificación no fue creada, desea volver a intentarlo?", "SATI", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                CrearNotificacionActualizarCliente.idActualizacion = idActualizacion
                'CrearNotificacionDescuentoPromesa.tipoCredito = TipoCredito
                CrearNotificacionActualizarCliente.ShowDialog()
                If creada Then
                    Task.Factory.StartNew(Sub()
                                              Dim comandoActualizaPromesa As SqlCommand
                                              Dim consultaActualizaPromesa As String
                                              consultaActualizaPromesa = "update ActualizacionesClientes set notificado = 1 where id = '" & idActualizacion & "'"
                                              comandoActualizaPromesa = New SqlCommand
                                              comandoActualizaPromesa.Connection = conexionempresa
                                              comandoActualizaPromesa.CommandText = consultaActualizaPromesa
                                              comandoActualizaPromesa.ExecuteNonQuery()


                                          End Sub)
                    Me.Close()
                Else
                    MessageBox.Show("La notificación no fue creada, puedes intentarlo más tarde")
                    Me.Close()

                End If
            Else
                Me.Close()
            End If

        End If
    End Sub
End Class